doc/cookbook/source/examples/classifier/hierarchical.rst

```diff
@@ -0,0 +1,36 @@
+=======================
+Hierarchical Clustering
+=======================
+
+Hierarchical clustering is a method of cluster analysis which seeks to build a hierarchy of clusters.
+In Shogun, we applies "bottom up" approach for the hierarchical clustering: each observation starts in its own cluster, and pairs of clusters are merged as one moves up the hierarchy.
+
+The merges are determined in a greedy manner. 
+To start with, we construct a distance matrix, where it saves the distance between the i-th and j-th elements.
```

Comment:
watch your English :)
Use something as "We start by constructing a pairwise distance matrix. Then, the clusters of the pair with closest distance are merged iteratively."